Aplib.Core/Belief/Belief.cs

```diff
@@ -56,26 +60,28 @@ public Belief(TReference reference, Func<TReference, TObservation> getObservatio
         /// <param name="reference">The object reference used to generate/update the observation.</param>
         /// <param name="getObservationFromReference">A function that takes an object reference and generates/updates an observation.</param>
         /// <param name="shouldUpdate">A condition on when the observation should be updated.</param>
-        public Belief(TReference reference, Func<TReference, TObservation> getObservationFromReference, Func<bool> shouldUpdate)
+        public Belief(TReference reference,
+            Func<TReference, TObservation> getObservationFromReference,
+            Func<bool> shouldUpdate)
             : this(reference, getObservationFromReference)
         {
             _shouldUpdate = shouldUpdate;
         }
 
         /// <summary>
-        /// Implicit conversion operator to allow a <see cref="Belief{TReference, TObservation}"/> object 
+        /// Implicit conversion operator to allow a <see cref="Belief{TReference, TObservation}"/> object
         /// to be used where a <typeparamref name="TObservation"/> is expected.
         /// </summary>
         /// <param name="belief">The <see cref="Belief{TReference, TObservation}"/> object to convert.</param>
-        public static implicit operator TObservation(Belief<TReference, TObservation> belief) => belief._observation;
+        public static implicit operator TObservation(Belief<TReference, TObservation> belief) => belief.Observation;
```

Comment:
Have you considered removing the implicit conversion, now that we have a public getter?

Reply:
This conversion is meant as a quality of life method (the public observation property as well). I don't think it is meant to be removed with the addition of the public property.

Reply:
Forcing the user to write `.Observation` results in clearer code, but also having the implicit conversion may be nice. I don't have a strong opinion, just wanted to check that it was considered.

Reply:
You are absolutely right. However, removing the implicit conversion would result in a breaking change, meaning that all aplib tests wouldn't even compile. 

We must try to avoid a breaking change at all cost, but we could make it deprecated.

Reply:
True, I think we can keep it (un-deprecated), unless it ever starts causing issues with type checking.